.github/workflows/e2e_tests.yaml

```diff
@@ -0,0 +1,56 @@
+name: End-to-End Depthai-nodes tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      additional-parameter:
+        description: 'Additional parameter: -all or -p <parser_name>. Default: -all. If -p is used: -p <parser_name>'
+        required: true
+        default: '-all'
+
+jobs:
+  HIL-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: [3.12]
```

Comment:
3.12 is used because hil_framework deps are not supported in 3.8. In the GH runner, python is only used to reserve the testbed and ssh to testbed server. The tests are performed on testbed server.